### ISSUE TITLE
Does not work with Ruby 3.0.0? (No such file or directory @ rb_sysopen)

### ISSUE THREAD

**Issue by pro-sumer** [tpope/rumember#18]
While `rumember` worked for me with Ruby 2.7.1, I get this error on Ruby 3.0.0:

```
/Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:134:in `initialize': No such file or directory @ rb_sysopen - http://api.rememberthemilk.com/services/rest?method=rtm.auth.getFprosumer&format=json&api_key=<REDACTED>&v=2&api_sig=<REDACTED> (Errno::ENOENT)
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:134:in `open'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:134:in `dispatch'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:88:in `authenticate'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:97:in `reconfigure'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:114:in `account'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:124:in `account'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:42:in `run'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/bin/ru:6:in `<top (required)>'
	from /Users/prosumer/.rbenv/versions/3.0.0/bin/ru:23:in `load'
	from /Users/prosumer/.rbenv/versions/3.0.0/bin/ru:23:in `<main>'
```

**Comment by pro-sumer** [comment]
@tpope Thank you! Can you please release this fix, so that we can use `gem install` to get this version?

For now I manually applied your change from commit fbe8816, but then I still get an error:

```
/Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/json-1.8.6/lib/json/common.rb:155:in `initialize': wrong number of arguments (given 2, expected 1) (ArgumentError)
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/json-1.8.6/lib/json/common.rb:155:in `new'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/json-1.8.6/lib/json/common.rb:155:in `parse'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:135:in `dispatch'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:88:in `authenticate'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:97:in `reconfigure'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:114:in `account'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:124:in `account'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/lib/rumember.rb:42:in `run'
	from /Users/prosumer/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rumember-1.1.0/bin/ru:6:in `<top (required)>'
	from /Users/prosumer/.rbenv/versions/3.0.0/bin/ru:23:in `load'
	from /Users/prosumer/.rbenv/versions/3.0.0/bin/ru:23:in `<main>'
```

**Comment by tpope** [comment]
Well we definitely don't want to cut a release if it isn't working. See if changing the json dependency to `~> 2.0` works.

**Comment by pro-sumer** [comment]
I'm sorry but I don't know enough about Ruby/gems to do that.

What should I do after changing the dependency in `rumember.gemspec`?

(to prevent this error: `Could not find 'json' (~> 1.8.0)`; a recursive grep does not show `1.8.0` anywhere)

**Comment by tpope** [comment]
It's in `rumember.gemspec`. Use `rake install` to build the gem and reinstall it.

**Comment by pro-sumer** [comment]
After changing `1.8.0` into `2.0` in both `rumember.gemspec` and `rumember-1.1.0.gemspec` (in 2 places in the latter!) it seems to work.

**Comment by tpope** [comment]
Released 1.1.1.
